docs/contribution/release.md

```diff
@@ -0,0 +1,78 @@
+# how release works
+
+Prerequisite:
+
+* docker installed
+* How drone works
+* How [goreleaser](https://goreleaser.com/intro/) works
+* multi arch TODO, don't know yet
+
+Briefly at least.
+
+## Continuous Delivery
+
+We use drone as continuous integration and continuous delivery pipeline.
+Integration means: running tests, checking code quality.
+Delivery means: make a release when a new tag get pushed.
+
+## goreleaser
+
+GoReleaser is a popular tool to release Go applications and library.
+
+I use it a lot because it is very flexible and it supports a lot of features, we
+need only a subset of them:
+
+* Multi arch build. It uses Go cross compilation feature to build for Darwin and
+  Linux (arm and amd)
+* Docker to build docker images for Linux container arm and amd.
+* Changelog to generate a changelog that will be pushed in the GitHub release
+  page with the list of PR part of the release itself.
+* Artifact (binaries) will be pushed as part of GitHub release page as well as
+  the other YAML file required by cluster-api like: metadata.yaml,
+  infrastructure-provider.yaml and so on.
+
+## Directories
+
+The release life cycle touches two directories:
+
+1. `./dist` it is a temporary directory used by gorleaser to bundle all the
+   required files and binaries as long as the generated changelog and so on. It
+   does not need to be checked in git and that's why it is in the gitignore.
+2. `./out` is ignored as well and it gets generated via `make release`. It is a
+   bundle that contains generated Kubernetes manifest required by the
+   cluster-api such as: metadata.yaml, infrastructure-provider.yaml,
+   cluster-template.yaml and so on.
```

Comment:
How are they bundled together? Does goreleaser create them all and move them all into `dist/` and then create an actual release? Or does it use both directories in parallel?

Reply:
I can mention looking at the `.goreleaser.yaml` file in the root of the project because it gives a good context of what goreleaser does even if you don't know it

Goreleaser builds binary by its own in `dist` and it gets files from the `out` directory (generated via the command you did `make release`)

if you look at the `.goreleaser.yaml` file you will see how goreleaser includes them from the `out` directory generated with the `make releaser` command

Reply:
Got it. Can we add that to the doc? Something as simple as: the details are in goreleaser.yaml, but the process goreleaser does is:

1. build binaries in dist
1. make output artifacts in out/ (we had better make sure those exist first)
1. combine them all in dist
1. create a release